spipe/main.c

```diff
@@ -17,15 +17,25 @@
 #include "pushbits.h"
 
 static int
-callback_conndied(void * cookie)
+callback_conndied(void * cookie, int reason)
 {
 	int * conndone = cookie;
 
+	/* Check reason. */
+	if (reason == PROTO_CONN_CONNECT_FAILED) {
```

Comment:
This seems like it should be a `switch`.

Reply:
err, really?  This seems unnecessarily verbose:

```
        /* Check reason. */
        switch (reason) {
        case PROTO_CONN_CONNECT_FAILED:
                connection_failed = 1;
                break;
        default:
                ;
                /* FALLTHROUGH */
        }
```

Reply:
Well, it depends if you're going to do anything else with the reason codes later.  I was guessing that you probably would, since otherwise why have multiple codes which are handled identically?

Reply:
Well, the git commit message was:

Currently, most errors are the generic PROTO_CONN_ERROR, but if there's
a desire to distinguish other cases (such as timeouts vs. protocol
handshake errors), that's easily added.

but I could rephrase that to say "Currently, the only one that will receive special attention is PROTO_CONN_CONNECT_FAILED"?

Reply:
The commit message is fine.  I just thought that it would make sense to write this code in a way which makes future progress easier.